src/utils/options/normalizeInputOptions.ts

```diff
@@ -53,7 +54,7 @@ export function normalizeInputOptions(config: InputOptions): {
 		moduleContext: getModuleContext(config, context),
 		onwarn,
 		perf: config.perf || false,
-		plugins: ensureArray(config.plugins),
+		plugins: ensureArray(config.plugins) as Plugin[],
```

Comment:
Is that an improvement?

Reply:
> Is that an improvement?

depends on the point of view. 🤣  currently, maybe not so much. when I cruise thru the code I see a lot of type assertions and non-null assertions (!), where I feel like I want to fix those. the problem is that no matter which path you go you always find yourself in a rabbit hole 😉  so I usually revert and let it be.

the current ensure array typing is incorrect in the sense that the generic <T> already covers `false | undefined | null`. I believe it's just there to make the type checker happy, but the root problem is elsewhere.

that's why I left this PR as a draft, I'll have a couple questions when I get back to it.

Reply:
Sure 😉, I really like that you look through the code  search for ways to make it better!